File: Editorial/137_Single_Number_II/CSharp/2_Hash_Map.cs
public class Solution {
    public int SingleNumber(int[] nums) {
        Dictionary<int, int> freq = new Dictionary<int, int>();
        foreach (int num in nums) {
            if (!freq.ContainsKey(num)) {
                freq[num] = 1;
            } else {
                freq[num]++;
            }
        }

        int loner = 0;
        foreach (KeyValuePair<int, int> entry in freq) {
            if (entry.Value == 1) {
                loner = entry.Key;
                break;
            }
        }

        return loner;
    }
}

File: Editorial/137_Single_Number_II/TypeScript/2_Hash_Map.ts
function singleNumber(nums: number[]): number {
    let freq: { [key: number]: number } = {};
    for (let num of nums) {
        if (freq[num] === undefined) {
            freq[num] = 1;
        } else {
            freq[num]++;
        }
    }

    let loner = 0;
    for (let key in freq) {
        if (freq[key] === 1) {
            loner = parseInt(key);
            break;
        }
    }

    return loner;
}
